lib/PostEditorBlocks/MarketingOptinBlock.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace MailPoet\PostEditorBlocks;
+
+use Automattic\WooCommerce\Blocks\Integrations\IntegrationInterface;
+use MailPoet\Config\Env;
+
+/**
+ * Class MarketingOptinBlock
+ *
+ * Class for integrating marketing optin block with WooCommerce Checkout.
+ */
+class MarketingOptinBlock implements IntegrationInterface {
+  /** @var array */
+  private $options;
+
+  public function __construct(
+    array $options
+  ) {
+    $this->options = $options;
+  }
+
+  /**
+   * The name of the integration.
+   *
+   * @return string
+   */
+  public function get_name() { // phpcs:ignore
+    return 'mailpoet';
```

Comment:
Can you please explain where is this used? Should it be capitalised maybe? `MailPoet`

Reply:
`get_name` is used in several places, and is enqueued in the frontend as well, so the standard is to use snake_case.

Reply:
This is the integration name, part of `IntegrationInterface`, and needs to be unique per extension. @senadir does case matter?

Reply:
name is used an array key and is exposed with settings data as `$name_data`. There is no requirement for it to be snake_case, but it's standard usage across plugins.

Reply:
Thanks for explaining!